architecture/2024-11-ucp-component-threat-model.md

```diff
@@ -0,0 +1,247 @@
+# Radius UCP Component Threat Model
+
+- **Author**: ytimocin
+
+## Overview
+
+This document provides a threat model for the Radius UCP (Universal Control Plane) component. It identifies potential security threats to this critical part of Radius and suggests possible mitigations. The document includes an analysis of the system, its assets, identified threats, and recommended security measures to protect the system.
+
+One of the most important goals of Radius is to be cloud agnostic and to enable portability of applications across platforms and environments. Therefore, the Radius UCP (Universal Control Plane) component helps the user apply ARM capabilities to non-Azure hosting models like AWS, GCP, and Kubernetes.
+
+In brief, UCP is a lightweight proxy that stands in front of the resource providers like Applications RP, AWS, and Azure. Based on the requests UCP gets, it forwards those requests to the appropriate service.
+
```

Comment:
I'd expect to see a few things highlighted here:

- UCP may have credentials for Azure or AWS, and may use those credentials to manage resources on the user's behalf.
- UCP is responsible for routing HTTP traffic from clients to each destination (resource provider). In the future this will include RBAC and authentication. Right now there is no RBAC and traffic between UCP and RPs is unauthenticated.